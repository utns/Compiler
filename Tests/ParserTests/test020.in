var
    a, i: integer;
    
procedure foo();
var
    i: integer;
begin
    i := 10;
    while i > 0 do
    begin
        i := i - 1;
        a := i * a;
    end;
end;

begin
    a := 1; 
    for i := 1 to 15 do
        foo();
end.

