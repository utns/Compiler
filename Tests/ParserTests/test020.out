begin
___10
:=
____i
while
     ____0
____>
     ____i
begin
          ____1
     ____-
          ____i
___:=
     ____i
          ____a
     ____*
          ____i
___:=
     ____a
__end
end

begin
____1
:=
____a
for i
____1
to
___15
____(
     __foo
end
